src/views/clusteroverview/OverviewTab/metric-charts-card/utils/hooks/useXAxisTicks.ts

```diff
@@ -0,0 +1,66 @@
+import { dateFormatterNoYear } from '@kubevirt-utils/components/Timestamp/utils/datetime';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+
+import { getLabeledTickIndexes } from '../utils';
+
+import { UseXAxisTicks } from './types';
+import { getPointDay } from './utils';
+
+// Ticks should be placed at the center point of each day's data
+const useXAxisTicks: UseXAxisTicks = (chartData) => {
+  const { t } = useKubevirtTranslation();
+  if (!chartData) {
+    return [null, null];
+  }
+
+  const getPrevDay = (pointDay: number): number => (pointDay === 0 ? 6 : pointDay - 1);
+
+  const isSingleDayData = (): boolean =>
+    getPointDay(chartData?.[0]) === getPointDay(chartData?.[chartData?.length - 1]);
+
+  // Get the index for the start and end of each day in the data
+  const getDayBoundaryIndexes = (): { [key: string]: { start: number; end: number } } =>
+    chartData.reduce(
+      (acc, point, idx) => {
+        const pointDay = getPointDay(point);
+
+        if (!acc.hasOwnProperty(pointDay)) {
+          acc[pointDay] = { start: idx, end: -1 };
+          acc[getPrevDay(pointDay)]['end'] = idx - 1;
+        }
+
+        if (idx === chartData.length - 1) {
+          acc[pointDay]['end'] = idx;
+        }
+
+        return acc;
+      },
+      { [getPointDay(chartData[0])]: { start: 0, end: -1 } },
+    );
```

Comment:
i understand those calculations needed, and usually, I'm not upvoting for comments n the codes, but in this case, maybe add some description? as in first sight seems a bit complicated to me

Reply:
Fixed!